crates/forge/src/cheatcodes_hint_processor.rs

```diff
@@ -218,7 +221,37 @@ impl CairoHintProcessor<'_> {
 
                 self.cheated_state.stop_prank(contract_address)
             }
-            "mock_call" => todo!(),
+            "start_mock_call" => {
+                let contract_address = ContractAddress(PatriciaKey::try_from(StarkFelt::new(
+                    inputs[0].clone().to_be_bytes(),
+                )?)?);
+                let fn_name = inputs[1].clone();
+                let fn_name = as_cairo_short_string(&fn_name)
+                    .unwrap_or_else(|| panic!("Failed to convert {fn_name:?} to Cairo short str"));
+                let fn_name = selector_from_name(fn_name.as_str());
+
+                let ret_data_length = inputs[2]
+                    .to_usize()
+                    .expect("Missing ret_data len in inputs");
+                let mut ret_data = vec![];
+                for felt in inputs.iter().skip(3).take(ret_data_length) {
+                    ret_data.push(felt_to_stark_felt(&felt.clone()));
+                }
+
+                self.cheated_state
+                    .start_mock_call(contract_address, fn_name, ret_data)
+            }
+            "stop_mock_call" => {
+                let contract_address = ContractAddress(PatriciaKey::try_from(StarkFelt::new(
+                    inputs[0].clone().to_be_bytes(),
+                )?)?);
+                let fn_name = inputs[1].clone();
+                let fn_name = as_cairo_short_string(&fn_name)
+                    .unwrap_or_else(|| panic!("Failed to convert {fn_name:?} to Cairo short str"));
+                let fn_name = selector_from_name(fn_name.as_str());
+
+                self.cheated_state.stop_mock_call(contract_address, fn_name)
+            }
```

Comment:
Other cheatcodes are now extracted to separate methods, this should be as well

Reply:
https://github.com/foundry-rs/starknet-foundry/blob/master/crates/forge/src/cheatcodes_hint_processor.rs#L202-L220
prank, roll, and warp also extract input arguments before passing them to separate methods, so this mock_call behaviour is consistent